### PR TITLE
feat: bias download

### PR DESCRIPTION
**Pull request by JustSimplyKyle**
need some testing. The logic seems to work, but there's a weird pause after the download supposedly complete

### CODE REVIEW

File: app/native/src/api.rs [hunk 1]
```diff
@@ -67,32 +67,45 @@ pub fn setup_logger() -> anyhow::Result<()> {
...
+
+    let vanilla_bias = DownloadBias {
+        start: 0.0,
+        end: 100.0,
+    };
+    run_download(stream, c.0, vanilla_bias).await?;
     launch_game(c.1.launch_args).await
 }
 
+pub struct DownloadBias {
```
Comment (SiongSng) [inline app/native/src/api.rs]: 建議可把這個 `struct` 移動到 downloads 的模塊（module）中，並簡單說明權重的使用方式，這樣未來重複使用時會方便許多

**Review by SiongSng — CHANGES_REQUESTED** [review]
:place_of_worship: :place_of_worship: :place_of_worship: 
:mobai: （~~昨天剛學到的~~）

File: app/native/src/api.rs (hunk 1 above)
Comment (JustSimplyKyle) [reply app/native/src/api.rs]: done! moved it to `download` and additionally the other stuffed are refactored in

File: app/native/src/api/vanilla/util.rs [hunk 2]
```diff
@@ -0,0 +1 @@
+
```
Comment (SiongSng) [inline app/native/src/api/vanilla/util.rs]: 這裡似乎沒東西

**Review by SiongSng — CHANGES_REQUESTED** [review]
剩這個就可以合併惹

File: app/native/src/api/vanilla/util.rs (hunk 2 above)
Comment (JustSimplyKyle) [reply app/native/src/api/vanilla/util.rs]: 啊這個忘記刪

**Review by SiongSng — APPROVED** [review]
LGTM

**Pull request merged**